test/unit/fiber.cc

```diff
@@ -374,6 +376,105 @@ cord_cojoin_test(void)
 	footer();
 }
 
+static void
+fiber_test_defaults()
+{
+	header();
+
+#ifndef NDEBUG
+	is(fiber_debug_enabled, true, "fiber_debug_enabled default");
+	is(fiber_abort_on_gc_leak, true, "fiber_abort_on_gc_leak default");
+#else
+	is(fiber_debug_enabled, false, "fiber_debug_enabled default");
+	is(fiber_abort_on_gc_leak, false, "fiber_abort_on_gc_leak default");
+#endif
+
+	footer();
+}
+
+static NOINLINE int
+leaker_f(va_list ap)
+{
+	region_alloc(&fiber()->gc, 1);
+	return 0;
+}
+
+static void
+fiber_test_leak(bool backtrace_enabled)
+{
+	header();
+
+	bool debug_save = fiber_debug_enabled;
+	fiber_debug_enabled = backtrace_enabled;
+	fiber_abort_on_gc_leak = false;
+
+	int fd = open("log.txt", O_RDONLY);
+	fail_if(fd == -1);
+	int rc = lseek(fd, 0, SEEK_END);
+	fail_if(rc == -1);
+
+	struct fiber *fiber = fiber_new_xc("leak", leaker_f);
+	fiber_set_joinable(fiber, true);
+	fiber_wakeup(fiber);
+	fiber_join(fiber);
+
+	fiber_abort_on_gc_leak = true;
+	fiber_debug_enabled = debug_save;
+
+	char buf[BUF_SIZE];
+	rc = read(fd, buf, BUF_SIZE - 1);
+	close(fd);
+	fail_if(rc == -1);
+	buf[rc] = '\0';
+
+#ifdef ENABLE_BACKTRACE
+	if (backtrace_enabled) {
+		const char *msg = "Fiber gc leak is found. "
+				  "First leaked fiber gc allocation"
+				  " backtrace:";
+		char *s = strstr(buf, msg);
+		fail_unless(s != NULL);
+		/*
+		 * Do not test for `region_alloc` frame as it is inlined
+		 * in release build.
+		 */
+		s = strstr(s, "leaker_f");
+		fail_unless(s != NULL);
+		s = strstr(s, "fiber_cxx_invoke");
+		fail_unless(s != NULL);
+	} else {
+		const char *msg =
+			"Fiber gc leak is found. "
+			"Leak backtrace is not available. "
+			"Make sure fiber.debug_enable() is called before"
+			" starting this fiber to obtain the backtrace.";
+		char *s = strstr(buf, msg);
+		fail_unless(s != NULL);
+	}
+#else
+	const char *msg =
+			"Fiber gc leak is found. "
+			"Leak backtrace is not available on your platform.";
+	char *s = strstr(buf, msg);
+	fail_unless(s != NULL);
+#endif
+
+	footer();
+}
+
+static void
+fiber_test_leak_modes()
+{
+	say_logger_init("log.txt", S_ERROR,
+			0 /* nonblock */, "plain");
+
+	fiber_test_leak(true /* backtrace_enabled */);
+	fiber_test_leak(false /* backtrace_enabled */);
+
+	say_logger_free();
+	unlink("log.txt");
```

Comment:
Nit: I don't think we need to delete the log file on failure, because it may be useful for investigation. Looks like other unit tests that create a log file don't bother to delete it. It should be fine, since when the test is started by test-run, the file will created in a test directory and deleted on the next test run.

Reply:
Done.